src/uwtools/rocoto.py

```diff
@@ -0,0 +1,46 @@
+"""
+Support for creating Rocoto XML workflow documents.
+"""
+
+from uwtools.config import YAMLConfig
+from uwtools.j2template import J2Template
+
+# Private functions
+
+
+def _add_jobname(tree: dict) -> None:
+    """
+    Add a "jobname" attribute to each "task" element in the given config tree.
+
+    :param tree: A config tree containing "task" elements.
+    """
+    for element, subtree in tree.items():
+        element_parts = element.split("_", maxsplit=1)
+        element_type = element_parts[0]
+        if element_type == "task":
+            # Use the provided attribute if it is present, otherwise use the name in the key.
+            task_name = element_parts[1]
+            tree[element]["jobname"] = subtree.get("attrs", {}).get("name") or task_name
+        elif element_type == "metatask":
+            _add_jobname(subtree)
+
+
+# Public functions
+def write_rocoto_xml(input_yaml: str, input_template: str, rendered_output: str) -> None:
+    """
+    Main entry point.
+
+    :param input_yaml: Path to YAML input file.
+    :param input_template: Path to input template file.
+    :param rendered_output: Path to directory to write rendered XML file.
+    """
+    # Make an instance of the YAMLConfig class.
+    # Pass the input yaml to return a dict.
```

Comment:
`yaml` -> `YAML`, as it's an acronym.

Also, the return type of `YAMLConfig` if a `YAMLConfig` object which -- as the typechecker made so painfully clear -- is not a `dict`. In fact, since it's pretty clear that the line just below is a call to an object constructor, it shouldn't be necessary to mention, in a comment, a return type: The return type is the class whose constructor is being called.

All considered, I don't think these two lines of comment are very useful, as the following line of code says essentially just that.